src/front/glsl/constants.rs

```diff
@@ -225,6 +233,48 @@ impl<'a> ConstantSolver<'a> {
                             span,
                         ))
                     }
+                    crate::MathFunction::Clamp => {
+                        let (value, width) = match (const0, const1.unwrap(), const2.unwrap()) {
+                            (
+                                &ConstantInner::Scalar {
+                                    width,
+                                    value: value0,
+                                },
+                                &ConstantInner::Scalar { value: value1, .. },
+                                &ConstantInner::Scalar { value: value2, .. },
+                            ) => (
+                                match (value0, value1, value2) {
+                                    (
+                                        ScalarValue::Sint(a),
+                                        ScalarValue::Sint(b),
+                                        ScalarValue::Sint(c),
+                                    ) => ScalarValue::Sint(a.max(b).min(c)),
+                                    (
+                                        ScalarValue::Uint(a),
+                                        ScalarValue::Uint(b),
+                                        ScalarValue::Uint(c),
+                                    ) => ScalarValue::Uint(a.max(b).min(c)),
+                                    (
+                                        ScalarValue::Float(a),
+                                        ScalarValue::Float(b),
+                                        ScalarValue::Float(c),
+                                    ) => ScalarValue::Float(a.max(b).min(c)),
+                                    _ => return Err(ConstantSolvingError::InvalidMathArg),
+                                },
+                                width,
+                            ),
+                            _ => return Err(ConstantSolvingError::InvalidMathArg),
```

Comment:
Generally, Naga IR follows the WGSL spec, and that says that the `clamp` function operates "component-wise when [the argument type] is a vector". It's fine not to bother handling vector arguments for now, but this should be a `NotImplemented` error, not an `InvalidMathArg` error.

Reply:
Done, though I haven't handled vector arguments yet.